CSRF programmatic API doesn't work

### Describe the bug

Using CSRF programmatic setup API from https://github.com/quarkusio/quarkus/pull/49618. When I setup an app with programmatic CSRF setup, as described [here](https://quarkus.io/guides/security-csrf-prevention#csrf-prevention-programmatic-set-up). After starting the app it will fail with:
```
...
Caused by: java.lang.IllegalStateException: Please add an extension that provides a CSRF prevention feature, for example the Quarkus REST Cross-Site Request Forgery Prevention `quarkus-rest-csrf` extension
...
```
even though I have the extension `io.quarkus:quarkus-rest-csrf` added.

### Expected behavior

App will start without problem.

### Actual behavior

App crashes during start.

### How to Reproduce?

Reproducer - https://github.com/mocenas/reproducer-quarkus-csrf-not-working
Try to run it with `quarkus dev`

Reproducer created from guide https://quarkus.io/guides/security-csrf-prevention just by steps:
```
quarkus create app org.acme:security-csrf-prevention \
    --extension='rest-csrf' \
    --no-code
cd security-csrf-prevention
```
Add `csrfToken.html` and `UserNameResource`.
Add class [CsrfProgrammaticConfig](https://quarkus.io/guides/security-csrf-prevention#csrf-prevention-programmatic-set-up).
Run with `quarkus dev`

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17

### Quarkus version or git rev

3.30.1  (or also 999-SNAPSHOT)

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/docs/src/main/asciidoc/security-csrf-prevention.adoc
+++ b/docs/src/main/asciidoc/security-csrf-prevention.adoc
@@ -329,7 +329,7 @@ include::{generated-dir}/config/quarkus-rest-csrf.adoc[leveloffset=+1, opts=opti
 [[csrf-prevention-programmatic-set-up]]
 == Configuring the CSRF prevention programmatically
 
-When the `quarkus-rest-csrf` extension is used, the `io.quarkus.vertx.http.security.HttpSecurity` CDI event allows you to customize the CSRF prevention programmatically:
+When the `quarkus-rest-csrf` extension is used together with the `quarkus-security` extension, the `io.quarkus.vertx.http.security.HttpSecurity` CDI event allows you to customize the CSRF prevention programmatically:
 
 [source,java]
 ----

--- a/extensions/resteasy-reactive/rest-csrf/deployment/src/test/java/io/quarkus/csrf/reactive/ProgrammaticCsrfValidationFailureTest.java
+++ b/extensions/resteasy-reactive/rest-csrf/deployment/src/test/java/io/quarkus/csrf/reactive/ProgrammaticCsrfValidationFailureTest.java
@@ -1,0 +1,23 @@
+package io.quarkus.csrf.reactive;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.vertx.http.security.CSRF;
+
+public class ProgrammaticCsrfValidationFailureTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest().withEmptyApplication();
+
+    @Test
+    public void testQuarkusSecurityExtensionRequired() {
+        var exception = assertThrows(IllegalStateException.class, CSRF::builder);
+        assertTrue(exception.getMessage().contains("Please add the `quarkus-security` extension"));
+    }
+
+}

--- a/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/HttpSecurityProcessor.java
+++ b/extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/HttpSecurityProcessor.java
@@ -303,21 +303,29 @@ public class HttpSecurityProcessor {
     @BuildStep
     void prepareCsrfConfigBuilder(Capabilities capabilities, Optional<CsrfBuilderClassBuildItem> csrfBuilderClassBuildItem,
             BuildProducer<BytecodeTransformerBuildItem> bytecodeTransformerProducer) {
-        if (capabilities.isPresent(Capability.SECURITY) && csrfBuilderClassBuildItem.isPresent()) {
+        if (csrfBuilderClassBuildItem.isPresent()) {
             final Class<? extends CSRF.Builder> csrfBuilderClass = csrfBuilderClassBuildItem.get().csrfBuilderClass;
-            // static Builder builder() {
-            //     return new io.quarkus.something.CsfrBuilder();
-            // }
-            bytecodeTransformerProducer.produce(new BytecodeTransformerBuildItem(CSRF.class.getName(), (cls, classVisitor) -> {
-                var classTransformer = new ClassTransformer(cls);
-                classTransformer.removeMethod("builder", CSRF.Builder.class);
-                try (var mc = classTransformer.addMethod("builder", CSRF.Builder.class)) {
-                    mc.setModifiers(ACC_PUBLIC | ACC_STATIC);
-                    var builderInstance = mc.newInstance(MethodDescriptor.ofConstructor(csrfBuilderClass));
-                    mc.returnValue(mc.checkCast(builderInstance, CSRF.Builder.class));
-                }
-                return classTransformer.applyTo(classVisitor);
-            }));
+            bytecodeTransformerProducer
+                    .produce(new BytecodeTransformerBuildItem(CSRF.class.getName(), (cls, classVisitor) -> {
+                        var classTransformer = new ClassTransformer(cls);
+                        classTransformer.removeMethod("builder", CSRF.Builder.class);
+                        try (var mc = classTransformer.addMethod("builder", CSRF.Builder.class)) {
+                            mc.setModifiers(ACC_PUBLIC | ACC_STATIC);
+                            if (capabilities.isPresent(Capability.SECURITY)) {
+                                // static Builder builder() {
+                                //     return new io.quarkus.something.CsfrBuilder();
+                                // }
+                                var builderInstance = mc.newInstance(MethodDescriptor.ofConstructor(csrfBuilderClass));
+                                mc.returnValue(mc.checkCast(builderInstance, CSRF.Builder.class));
+                            } else {
+                                // static Builder builder() {
+                                //     throw new IllegalStateException("Please add the `quarkus-security` extension");
+                                // }
+                                mc.throwException(IllegalStateException.class, "Please add the `quarkus-security` extension");
+                            }
+                        }
+                        return classTransformer.applyTo(classVisitor);
+                    }));
         }
     }
 

